wrappers/node/src/api/logging.ts

```diff
@@ -1,16 +1,19 @@
 import * as ffi from 'ffi-napi'
 import * as ref from 'ref-napi'
-import * as Struct from 'ref-struct-di'
+import * as buildStructType from 'ref-struct-di'
 
 import { VCXInternalError } from '../errors'
 import { rustAPI } from '../rustlib'
 
+const Struct = buildStructType(ref)
+
 export const Logger = Struct({
   flushFn: ffi.Function('void', []),
   logFn: ffi.Function('void', ['int', 'string', 'string', 'string', 'string', 'int'])
 })
 
-export function loggerToVoidPtr (_logger: any | Struct): Buffer {
+// The _logger must in fact be instance of Struct type we generated above using buildStructType(ref)
+export function loggerToVoidPtr (_logger: any): Buffer {
```

Comment:
The type is constructed programatically as:
```
const Struct = buildStructType(ref)
```
so to be correct in signature I'd have to write 
```
export function loggerToVoidPtr (_logger: Struct): Buffer {
```
Typescript won't allow that because it doesn't think the `Struct` is actual type. It might be possible to get it done somehow but it doesn't seem straightforward. The code as was previously passed typescript checks but didn't work at all.